src/components/events/EventModal.tsx

```diff
@@ -0,0 +1,276 @@
+// Component for the Ajust Event form
+
+// TODOO copyed from adjust Event.
+import {
+    Button,
+    Checkbox,
+    FormControl,
+    FormHelperText,
+    FormLabel,
+    Input,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay,
+} from '@chakra-ui/react';
+// import { Sponsor } from 'model';
+import { EventType, GrowEvent } from 'model';
+import { useEffect, useState } from 'react';
+
+interface EventModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    onSave: (event: Omit<GrowEvent, 'id'>) => void;
+    onDelete: () => void;
+    initialValue: Omit<GrowEvent, 'id'> & { id?: number };
+    onCreate: (event: Omit<GrowEvent, 'id'>) => void;
+}
+
+// TODO: delete Commits
+// View Theme properties
+// Modal view type
+/*
+enum ModalViewType {
+    adjust = 'adjust',
+    create = 'create',
+}
+*/
+
+// Adjust:
+
+const adjustTitleText = 'Adjust Event';
+const createTitleText = 'Create New Event';
+
+const SavebuttonText = 'Save';
+const CreatebuttonText = 'Create';
+
+/*
+function getThemeText1(viewTheme: ModalViewType) {
+    switch (viewTheme) {
+        case ModalViewType.adjust:
+            return adjustTitleText;
+        case ModalViewType.create:
+            return createTitleText;
+    }
+}
+*/
+
+/**
+ * Get the Window Headline of the modal, if the event will create or adjust.
+ * @param id event id
+ * @returns if the id is undefine return create Title String, otherwiese return the adjust title String
+ */
+function getThemeText(id: number | undefined) {
+    if (isIdUndefined(id)) {
+        // id is empty -> create new Event
+        return createTitleText;
+    } else {
+        return adjustTitleText;
+    }
+}
+
+function getButtonText(id: number | undefined) {
+    if (isIdUndefined(id)) {
+        return CreatebuttonText;
+    } else {
+        return SavebuttonText;
+    }
+}
+
+/* 
+function getButtonAction(event) {
+    if (isIdUndefined(event.id)) {
+        return;
+    } else {
+        return () => onSave(event);
+    }
+}
+
+function isViewThemeEqual(
+    viewTheme1: ModalViewType,
+    viewTheme2: ModalViewType
+) {
+    if (viewTheme1 == viewTheme2) {
+        return true;
+    } else {
+        return false;
+    }
+}
+*/
+
+/**
+ * Check if the event id is valid
+ * @param id: the event id
+ * @returns true if the id is valid, otherwise false
+ */
+function isIdUndefined(id: number | undefined) {
+    if (id == undefined) {
+        return true;
+    } else {
+        return false;
+    }
+}
```

Comment:
ich frage mich, ob diese Funktion wirklich notwendig ist. Ob ich jetzt `isIdUndefined(id)` oder `id === undefined` verwende ist relativ egal. Aber für jede Variable eine Methode `is...Undefined` zu schreiben ist denke ich ziemlich viel overhead